core/src/server/subscription.rs

```diff
@@ -322,6 +322,11 @@ impl SubscriptionSink {
 		self.method
 	}
 
+	/// Get the connection ID.
+	pub fn connection_id(&self) -> ConnectionId {
```

Comment:
Amazing!

Different topic: Would it also be possible to identify the client connected by this `ConnectionID`? (https://github.com/paritytech/substrate/issues/14734)

Reply:
Yes that should work but the connection IDs may be reused once, the connection is dropped and the counter wraps.

I guess that could handled when the subscription is dropped then just remove it from a HashMap or something